### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by filmor**

### CODE REVIEW

**Comment by filmor** [comment]
@evnu fyi

File: Cargo.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [workspace]
+resolver = "2"
```
Comment (evnu) [inline Cargo.toml:2]: Should we split that and the change for edition 2021 into a separate PR? Or is it relevant to fixing the lints?

**Review by evnu — APPROVED** [review]
LGTM; Don't know if we should have an extra PR for the edition change to make it more explicit and add it to the changelog.

File: Cargo.toml (hunk 1 above)
Comment (filmor) [reply Cargo.toml:2]: It fixes another warning, not a "fatal" one, though. I'll move it to a separate PR (and merge it directly).

**Pull request merged**